### DISCUSSION TITLE
Blue bar beside active tab

### DISCUSSION THREAD

**Discussion by sdfg2** [piroor/treestyletab#3222] [Q&A] [answered]
I'm making my own theme from scratch, and one thing I cannot figure out is this blue bar beside active tabs:

![image](https://user-images.githubusercontent.com/5736478/194274738-99cf1ab3-34a8-4705-b88e-28103fde58f7.png)

(the screenshot is taken without any customizations applied).

I know it's somewhere, because other themes don't have it, but I cannot figure out how to get rid of it.

**Comment by irvinm** [comment, marked as answer]
Here you go:

```
tab-item .highlighter {
	display: none;
}
```

**Reply by sdfg2** [reply]
I keep forgetting you can combine elements and classes and stuff!

Now I just need to figure out how to put a marker somewhere when a tab title changes.

**Reply by irvinm** [reply]
I'm not sure there will be an easy way to detect when a title changes.  However, if you want to just do this when a tab becomes "unread" (new tab or if content of the page changes), you could do something like this:

```
tab-item.unread tab-item-substance {
  border-left: 3px solid yellow;
}
```

You could probably do the title monitoring as a helper addon for TST and monitor the tab titles and if they change modify some display aspect ... of course this would be much more work.

**Reply by sdfg2** [reply]
Actually I just discovered what would be perfect - I sent a tab from my phone to my desktop and there was a wonderful, perfect, little blue dot sitting underneath the favicon.  I inspected the DOM and found that the tab-item had the `attention` class applied which I hadn't seen before.  I tweaked the classes of another tab-item to add `attention` and lo and behold, that worked.

Is there anyway to add `attention` to a tab if it's marked as `unread`?  Or duplicate the visual effect of the `attention` class for the `unread` class?

**Reply by irvinm** [reply]
You can probably simplify, but I just looked at base.css in the source and looked for tab-item.attention related items and switched them to unread items.  It seemed to work, but didn't test it too much.  (I don't think there is a way to have CSS apply class attributes ...)

```
tab-item.unread tab-favicon::before {
  background-image: var(--attention-marker-gradient);
}

tab-item.unread tab-favicon::before {
  bottom: 0;
  left: 0;
  pointer-events: none;
  position: absolute;
  right: 0;
  top: 0;
}

tab-item.unread tab-favicon::before {
  bottom: auto;
  content: '';
  height: var(--tab-current-size);
  left: var(--attention-marker-x-offset);
  right: auto;
  top: var(--attention-marker-y-offset);
  width: var(--tab-current-size);
}
```

**Reply by sdfg2** [reply]
You are an angel!  I would have never figured that out.  I didn't realise that there _was_ a base css, the preferences page when you select 'no theme' makes it seem like you have to do absolutely _everything_ from scratch.
